clang/lib/CIR/CodeGen/CIRGenRecordLayoutBuilder.cpp

```diff
@@ -177,18 +177,26 @@ void CIRRecordLowering::lower() {
     return;
   }
 
-  if (isa<CXXRecordDecl>(recordDecl)) {
-    cirGenTypes.getCGModule().errorNYI(recordDecl->getSourceRange(),
-                                       "lower: class");
-    return;
-  }
-
   assert(!cir::MissingFeatures::cxxSupport());
 
   CharUnits size = astRecordLayout.getSize();
 
   accumulateFields();
 
+  if (auto const *cxxRecordDecl = dyn_cast<CXXRecordDecl>(recordDecl)) {
```

Comment:
```suggestion
  if (const auto *cxxRecordDecl = dyn_cast<CXXRecordDecl>(recordDecl)) {
```
I am team east const personally, but we use const west throughout the CIR codebase already (unfortunately) :)